<launch>

  <!-- amcl -->
  <arg name="amcl" default="false"/>
  <node pkg="amcl" type="amcl" name="amcl" if="$(arg amcl)">
    <param name="min_particles"             value="100"/>
    <param name="max_particles"             value="1000"/>
    <param name="kld_err"                   value="0.08"/>
    <param name="update_min_d"              value="0.2"/>
    <param name="update_min_a"              value="0.2"/>
    <param name="resample_interval"         value="2.0"/>
    <param name="transform_tolerance"       value="0.1"/>
    <param name="recovery_alpha_slow"       value="0.0"/>
    <param name="recovery_alpha_fast"       value="0.0"/>
    <param name="gui_publish_rate"          value="10.0"/>
    <param name="use_map_topic"             value="false"/>
    <param name="laser_max_range"           value="10.0"/>
    <param name="laser_max_beams"           value="50"/>
    <param name="laser_z_hit"               value="0.5"/>
    <param name="laser_z_short"             value="0.05"/>
    <param name="laser_z_max"               value="0.05"/>
    <param name="laser_z_rand"              value="0.5"/>
    <param name="laser_sigma_hit"           value="0.2"/>
    <param name="laser_lambda_short"        value="0.1"/>
    <param name="laser_likelihood_max_dist" value="1.0"/>
    <param name="laser_model_type"          value="likelihood_field"/>
    <param name="odom_model_type"           value="diff"/>
    <param name="odom_alpha1"               value="0.01"/>
    <param name="odom_alpha2"               value="0.01"/>
    <param name="odom_alpha3"               value="0.01"/>
    <param name="odom_alpha4"               value="0.01"/>
  </node>

  <!-- fake_localization -->
  <node pkg="fake_localization" type="fake_localization" name="amcl" unless="$(arg amcl)">
    <remap from="base_pose_ground_truth"    to="odom"/>
    <param name="delta_x"                   value="0.0"/>
    <param name="delta_y"                   value="0.0"/>
    <param name="delta_yaw"                 value="0.0"/>
  </node>

  <!-- move base -->
  <arg name="move_base" default="true"/>
  <!-- <arg name="cmd_vel_topic" value="mobile_base/commands/velocity"/>  -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" if="$(arg move_base)">
    <rosparam file="$(find tms_rc_ninebot)/param/costmap_common_params.yaml" command="load" ns="global_costmap"/>
    <rosparam file="$(find tms_rc_ninebot)/param/costmap_common_params.yaml" command="load" ns="local_costmap"/>
    <rosparam file="$(find tms_rc_ninebot)/param/costmap_individual_params.yaml" command="load"/>
    <rosparam file="$(find tms_rc_ninebot)/param/base_local_planner_params.yaml" command="load"/> 
    <rosparam file="$(find tms_rc_ninebot)/param/move_base_params.yaml" command="load"/>
    <!-- <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/> -->
  </node>

</launch>
